docs/Storage Rules.md

```diff
@@ -0,0 +1,220 @@
+
+``` yaml
+functions:
+  isAuthenticated: 'return !!request.auth'
+  isOwner: "return !!request.auth && userId === request.auth['user-id']"
+  validToken: 'return request.query.token === resource.Metadata.token'
+paths:
+  /user/:userId/:
+    list: 'isOwner(userId)'
+  /user/:userId/:fileId:
+    read: 'isOwner(userId) || validToken()'
+    write: 'isOwner(userId)'
+```
+
+The `yaml` file is split into the following sections:
+
+ - [Paths](#paths)
+ - [Functions](#functions)
+
+---
+
+## Paths
+
+Paths allow you to define authorization permissions to your folders and files. This means you can control access to files, folders, and subfolders easily.
+
+Paths can be static or dynamic, and dynamic paths can be used as variables within functions.
+
+### Folder paths
+
+Folder paths
+
+``` yaml
+paths:
+  /user/:userId/files/:
+    # Rules
+```
+
+**Note the trailing slash**. This is how you define folder paths.
+
+You can use this to add permissions to listing files in a directory or downloading a `.zip` file of all the files.
+
+### File paths
+
+File paths define rules for the individual files within your storage.
+
+``` yaml
+paths:
+  /user/:userId/files/:fileId:
+    # Rules
+```
+
+Here, the `/user` and `/files` parts are static paths. The `:userId` and `:fileId` parts are dynamic paths.
+
+Here's an example path that would be validated by this rule:
+
+``` txt
+/user/1/files/image.png
+```
+
+## Rules
+
+You can specify the following rules in your `rules.yaml` file.
+
+| Action           | Metadata (`/m/`)                  | Object (`/o/`)                        |
+|------------------|-----------------------------------|---------------------------------------|
+| Folder: `update` | N/A                               | N/A                                   |
+| Folder: `list`   | Get metadata for accessible files | Get `.zip` folder of accessible files |
+| Folder: `get`    | N/A                               | N/A                                   |
+| Folder: `delete` | N/A                               | Delete folder and children files      |
+| File: `create`   | N/A                               | Create file                           |
+| File: `update`   | Update metadata                   | Update file                           |
+| File: `list`     | N/A                               | N/A                                   |
+| File: `get`      | Get file metadata                 | Get file                              |
+| File: `delete`   | N/A                               | Delete the file                       |
+
+For simple allow/deny, you can return boolean values. This should be a simple javascript expression in a string.
+
+``` yaml
+paths:
+  /public:
+    list: 'return true'
```

Comment:
list: 'true'

Reply:
Great, thanks. should it still be a string?

Reply:
Yes